cohttp-eio/src/cohttp_eio.mli

```diff
@@ -0,0 +1,158 @@
+(** [Reader] is a buffered reader. *)
+module Reader : sig
+  type t
+
+  val create : ?buffer_size:int -> #Eio.Flow.read -> t
+  (** [create ?buffer_size reader] creates [t]. [buffer_size] is the maximum
+      number of bytes [reader] attempts to read in one call. If [buffer_size] is
+      not given then [default_io_buffer_size] is used. *)
+
+  val reader : t -> Eio.Flow.read
+  (** [reader t] returns the reader used by [t]. *)
+
+  val buffer_size : t -> int
+  (** [bufer_size t] returns the current [reader t] read buffer size. *)
+
+  val default_io_buffer_size : int
+  (** [default_io_buffer_size] is [4096]. *)
+
+  (** {1 Low Level API} *)
+
+  val consume : t -> int -> unit
+  (** [consume t n] marks [n] bytes of data as consumed in [t]. *)
+
+  val feed_input : t -> Cstruct.t
+  (** [feed_input t] is [buf]. Attempts to read at most [buffer_size t] bytes
+      into [t] and returns a view into unconsumed buffer represented by [buf].
+      [buf.len = 0] if [reader t] has reached end of file. *)
+
+  (** {2 High Level API} *)
+
+  val read_into : t -> off:int -> len:int -> Cstruct.buffer -> int
+  (** [read_into t ~off ~len buf] fills [buf] from [off] to length [len] with
+      data from [t]. *)
+
+  val read_char : t -> char
+  (** [read_char t] reads and returns [char] from [t] or raises [End_of_file] if
+      [t] has reached end of input.*)
+end
+
+(** [Chunk] encapsulates HTTP/1.1 chunk transfer encoding data structures.
+    https://datatracker.ietf.org/doc/html/rfc7230#section-4.1 *)
+module Chunk : sig
+  type t =
+    | Chunk of {
+        data : Cstruct.t;
+        length : int;
+        extensions : chunk_extension list;
+      }
+    | Last_chunk of chunk_extension list
+
+  and chunk_extension = { name : string; value : string option }
+end
+
+(** [Request] is a HTTP/1.1 request. *)
+module Request : sig
```

Comment:
There are signatures for these modules in the http and cohttp packages, is there a reason you are re-defining them all?

Reply:
can we extend that instead? Like `include Cohttp.Request` and add any extra needed function? I am just thinking of ways to have as compatible interfaces as possible

Reply:
You could just provide one function that returns the underlying `Http.Request.t`. This would allow users to re-use whatever functions they have working on existing types.